lesson17/script.js

```diff
@@ -0,0 +1,147 @@
+const slider = document.getElementById("js-slider");
+
+const createElement = ({tagName,id,className,text}) => {
+    const element = document.createElement(tagName);
+    if(id){
+        element.id = id;
+    }
+    element.className = className;
+    element.textContent = text;
+    return element;
+}
+
+const createImage = ({id,className,src,alt}) => {
+    const image     = document.createElement("img");
+    if(id){
+        image.id = id;
+    }
+    image.className = className;
+    image.src = src;
+    image.alt = alt;
+    return image;
+}
+
+const createButton = ({id,className}) => {
+    const button = document.createElement("button");
+    button.id = id;
+    button.className = className;
+    return button;
+}
+
+const createErrorMessage = (error) => {
+    const errMsg  = createElement({tagName:"p",className:"error-msg",text:error})
+
+    slider.appendChild(errMsg);
+    return errMsg;
+}
+
+const showLoading = () => {
+    const loadingImg = createImage({
+        id:"js-loading",
+        className:"loading",
+        src:"img/loading-circle.gif",
+        alt:"ローディング"
+    });
+
+    slider.appendChild(loadingImg);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (URL) => {
+    const response = await fetch(URL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage('現在、サーバーの通信が壊れています');
+    }
+    const json = await response.json();
+    const data = json.data;
+    return data;
+}
+
+const fetchSliderData = (ms) => new Promise(resolve => setTimeout(() => resolve(fetchData("https://myjson.dit.upm.es/api/bins/2bzx")),ms));
+
+const init = async () => {
+    showLoading();
+    try{
+        const sliderData = await fetchSliderData(3000);
+        if (!sliderData) {
+            return;
+        }
+        renderSliderItems(sliderData);
+        renderPagenation(sliderData);
+        
+        renderButtons();
+        isActiveFirstSliderItem();
+        SwitchSliderItem();
```

Comment:
Q
Do you dare capitalize here?

Reply:
This is a simple mistake.
Modify to camel case.

Reply:
I fixed.
https://github.com/nakagawamai/js-lesson/pull/19/commits/0d1f52ab22b3ab33f76a2b18cd8be3e1651b2edf